app/transport/rest/user.go

```diff
@@ -96,13 +122,51 @@ func (uh UserHandler) GetUserByID() http.Handler {
 			return
 		}
 
-		user := User{
-			ID:        entUser.ID,
-			FirstName: entUser.FirstName,
-			LastName:  entUser.LastName,
-			Email:     entUser.Email,
-			CreatedAt: entUser.CreatedAt,
+		userResp := User{
+			ID:        user.ID,
+			FirstName: user.FirstName,
+			LastName:  user.LastName,
+			Email:     user.Email,
+			CreatedAt: user.CreatedAt,
 		}
-		WriteJSONResponse(w, http.StatusOK, user, uh.logger)
+		WriteJSONResponse(w, http.StatusOK, userResp, uh.logger)
+	})
+}
+
+// QueryParams represent request query params.
+type QueryParams struct {
+	Offset string `schema:"offset"`
+	Limit  string `schema:"limit"`
+	Sort   string `schema:"sort"`
+}
+
+// GetUsers retrieves all entities.User by given parameters.
+func (uh UserHandler) GetUsers() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		var params QueryParams
+
+		if err := schema.NewDecoder().Decode(&params, req.URL.Query()); err != nil {
+			uh.logger.Errorf("Failed parsing query %+v: %+v", req.URL.Query(), err)
+			WriteJSONResponse(w, http.StatusInternalServerError, Response{Message: MsgInternalSeverErr, Details: "could not parse query"}, uh.logger)
+
+			return
+		}
+
+		users, err := uh.usecase.GetAll(params.Offset, params.Limit, params.Sort)
```

Comment:
Func with this signature will be harder to support.
You can pass QueryParams instead of 3 specific arguments. In such a way you can extend or change QueryParams with minimal changes to the code itself.

It makes sense to make a shared QueryParams struct that can be reused by other endpoints as well.

Reply:
> It makes sense to make a shared QueryParams struct that can be reused by other endpoints as well.

I have already done it in the first iteration and was told it is too much for a single PR 😄

Reply:
You made to much refactoring stuff there.
In the very beginning it's better to split refactoring from new logic.
After some time, it's will be easier and will be able to start using boy scout rule.